scripts/process-reboot-cause

```diff
@@ -69,6 +71,84 @@ def read_reboot_cause_files_and_save_state_db():
                 x = TIME_SORTED_FULL_REBOOT_FILE_LIST[i]
                 os.remove(x)
 
+def get_dpus():
+    """Retrieve the list of DPUs from the platform.json file."""
+    dpus = []
+
+    try:
+        # Get platform information
+        platform_info = device_info.get_platform_info()
+        platform = platform_info.get('platform')
+        if not platform:
+            raise KeyError("Platform key missing from platform_info")
+
+        # Path to platform.json
+        platform_path = os.path.join("/usr/share/sonic/device", platform, "platform.json")
+        sonic_logger.log_info(f"Reading platform JSON from: {platform_path}")
+
+        with open(platform_path, 'r') as f:
+            platform_data = json.load(f)
+
+        dpus = platform_data.get('DPUS', [])
+        if not dpus:
+            sonic_logger.log_warning("No DPUS found in platform.json")
+
+    except Exception as e:
+        sonic_logger.log_error(f"Error retrieving DPUs: {e}")
+
+    return dpus
+
+def get_sorted_reboot_cause_files(dpu_history_path):
+    """Retrieve and sort the reboot cause files for a specific DPU."""
+    try:
+        # Assuming you have a way to list the files in the directory
+        files = os.listdir(dpu_history_path)
+        # Filter and sort the files based on your criteria (e.g., by modification time)
```

Comment:
@rameshraghupathy why eg? Are we not sure?

Reply:
@prgeor cleaned

Reply:
@rameshraghupathy I mean why eg? You are indeed sorting by modification time so just put "sorted by modification time"

Reply:
@prgeor cleaned